### PR TITLE
Feature bp status reassertion

### PR DESCRIPTION
**Pull request by mark-tsyrulnyk**
## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: contracts/contracts/eosio.system/src/voting.cpp [hunk 1]
```diff
@@ -378,4 +379,9 @@ namespace eosiosystem {
       );
    }
 
+   bool voter_info::should_reassert_bp_status() const {
```
Comment (roman-tik) [inline contracts/contracts/eosio.system/src/voting.cpp]: they reassert their vote, not the bp status

File: contracts/contracts/eosio.system/src/producer_pay.cpp [hunk 2]
```diff
@@ -76,8 +77,15 @@ namespace eosiosystem {
 
    using namespace eosio;
    void system_contract::claimrewards( const name& owner ) {
+      using namespace std::literals::string_literals;
```
Comment (olexiybuyanskyy) [inline contracts/contracts/eosio.system/src/producer_pay.cpp]: better to use std::string_literals

File: contracts/contracts/eosio.system/src/voting.cpp [hunk 3]
```diff
@@ -378,4 +379,8 @@ namespace eosiosystem {
       );
    }
 
+   bool voter_info::bp_status_active() const {
+         return current_time_point() - last_reassertion_time < microseconds( voter_info::reassertion_period * useconds_per_day );
```
Comment (olexiybuyanskyy) [inline contracts/contracts/eosio.system/src/voting.cpp]: Use parentheses
return (current_time_point() - last_reassertion_time) < microseconds( voter_info::reassertion_period * useconds_per_day );

https://wiki.sei.cmu.edu/confluence/display/c/EXP00-C.+Use+parentheses+for+precedence+of+operation

File: contracts/contracts/eosio.system/src/voting.cpp [hunk 4]
```diff
@@ -294,9 +294,10 @@ namespace eosiosystem {
       update_total_votepay_share( ct, -total_inactive_vpay_share, delta_change_rate );
 
       _voters.modify( voter, same_payer, [&]( auto& av ) {
-         av.last_vote_weight = new_vote_weight;
-         av.producers = producers;
-         av.proxy     = proxy;
+         av.last_vote_weight       = new_vote_weight;
```
Comment (olexiybuyanskyy) [inline contracts/contracts/eosio.system/src/voting.cpp]: I would not recommend changing the default style and would keep the existing one with no additional spaces

**Review by olexiybuyanskyy — APPROVED** [review]

File: contracts/contracts/eosio.system/include/eosio.system/eosio.system.hpp [hunk 5]
```diff
@@ -241,6 +241,10 @@ namespace eosiosystem {
     * - `staked` the amount staked
     */
    struct [[eosio::table, eosio::contract("eosio.system")]] voter_info {
+   public:
+      static constexpr auto reassertion_period = 7;
```
Comment (gBaGu) [inline contracts/contracts/eosio.system/include/eosio.system/eosio.system.hpp]: Maybe rename to reassertion_period_days to avoid confusion

**Review by me-vadim — APPROVED** [review]

**Pull request merged**